services/sqs/src/it/java/software/amazon/awssdk/services/sqs/SurfaceAreaReview.java

```diff
@@ -0,0 +1,377 @@
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.services.sqs.batchmanager.SqsAsyncBatchManager;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityBatchRequest;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityResponse;
+import software.amazon.awssdk.services.sqs.model.CreateQueueResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse;
+import software.amazon.awssdk.services.sqs.model.Message;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
+
+public class SurfaceAreaReview {
+
+
+    private static SqsAsyncClient globalSqsAsyncClient = SqsAsyncClient.create();
+    private static String QUEUE_URL = null;
+    ;
+
+
+
+    @Test
+    void creatingUsingAsyncClient() throws Exception {
+        if (QUEUE_URL == null) {
+            createTestQueue();
+        }
+        ApiCaptureInterceptor apiCaptureInterceptor = new ApiCaptureInterceptor();
+        SqsAsyncClient sqsAsyncClient =
+            SqsAsyncClient.builder().overrideConfiguration(o -> o.addExecutionInterceptor(apiCaptureInterceptor)).build();
+
+
+        //[REVIEW AREA  for batchManager from SqsAsyncClient]
+        SqsAsyncBatchManager sqsAsyncBatchManager = sqsAsyncClient.batchManager();
+
+
+        useCaseOfAPIs(sqsAsyncBatchManager);
+        apiCaptureInterceptor.callsMadeMap.forEach((key, value) -> System.out.println("Key: " + key + ", Value: " + value));
+        sqsAsyncBatchManager.close();
+    }
+
+    @Test
+    void creatingCustomConfigBatchMangers() throws Exception {
+        if (QUEUE_URL == null) {
+            createTestQueue();
+        }
+        ApiCaptureInterceptor apiCaptureInterceptor = new ApiCaptureInterceptor();
+
+
+        SqsAsyncClient sqsAsyncClient =
+            SqsAsyncClient.builder().overrideConfiguration(o -> o.addExecutionInterceptor(apiCaptureInterceptor)).build();
+
+        //[REVIEW AREA  for batchManager from SqsAsyncBatchManager Interface]
+
+        SqsAsyncBatchManager sqsAsyncBatchManager =
+            SqsAsyncBatchManager.builder()
+                                .client(sqsAsyncClient)
+                                .scheduledExecutor(Executors.newScheduledThreadPool(8))
+                                .overrideConfiguration(b -> b.minReceiveWaitTime(Duration.ofMillis(300))
+                                                             .maxDoneReceiveBatches(2)
+                                                             .adaptivePrefetching(true)
+                                                             .maxInflightReceiveBatches(2)
+                                                             .maxBatchItems(10))
```

Comment:
Should we flatten this down instead of having a separate override configuration?

Reply:
Those settings are not shared between different APIs right? If we keep them flattened, should we add the API prefix to the setting, e.g., receiveMaxBatchItems? Because it's a bit hard to tell which setting is for which API right now